src/types.ts

```diff
@@ -1,11 +1,24 @@
 import { TypedSlackAPIMethodsType } from "./typed-method-types/mod.ts";
 import { SlackAPIMethodsType } from "./generated/method-types/mod.ts";
+
 export type { DatastoreItem } from "./typed-method-types/apps.ts";
 
 export type {
   ValidTriggerTypes as Trigger,
 } from "./typed-method-types/workflows/triggers/mod.ts";
 
+// TODO: [brk-chg] remove this in favor of `Response`
+export type FullResponse = BaseResponse & {
+  /** Convert to the original `Response`. It returns a `Response`, that was successful.
+   *
+   * ```ts
+   * const originalResponse = response.toResponse();
+   * console.log(originalResponse.headers);
+   * ```
+   */
+  toFetchResponse(): Response;
+};
+
```

Comment:
TLDR: I think we should add `toFetchResponse()` to `BaseResponse` instead of creating a new `FullResponse` type.

I believe I saw somewhere that the point of introducing `FullResponse` is to avoid a breaking change to `BaseResponse`. Assuming that's the case, I don't think adding the `toFetchResponse` method constitutes a breaking change. If it was an argument, definitely, but since it's a response type it shouldn't be constructed by developers directly.

From what I remember we initially exported `BaseResponse` as a way for developers to type the API client, but now that we provide a type to represent the client `BaseResponse` is mostly useful for developers to leverage in their utility functions.

That means if we update our API calls to return `toFetchResponse` but don't update the `BaseResponse` directly, devs will need to update their utilities to use `FullResponse` to actually match return type of their API calls. That seems like a breaking change.

Reply:
Sounds good 💯  I will add `toFetchResponse()` to `BaseResponse` instead of creating `FullResponse`